src/main/python/id_contexts/up_student_card.py

```diff
@@ -0,0 +1,109 @@
+            'line_type': LineType.UNTITLED_ADJACENT,
+            'multi_line': False
+        }, {
+            'field': 'names',
+            'find': None,
+            'field_type': FieldType.TEXT_ONLY,
+            'to_uppercase': True,
+            'line_type': LineType.TITLED_NEWLINE,
+            'multi_line': False,
+        }]
+        # Initialise parent
+        IDContext.__init__(self, match_contexts)
+
+    def _dictify(self, match_contexts, id_string, barcode_data, fuzzy_min_ratio, max_multi_line):
+        """
+        This function is responsible for generating a dictionary object containing the relevant ID information,
+        such as names, surname, ID number, etc., from a given input string containing said relevant information.
+        In this particular ID context, the information is sparse and is mainly intended for demonstration purposes.
+
+        Authors:
+            Jan-Justin van Tonder
+
+        Args:
+            match_contexts (list): A list of dictionaries that contain the contextual information used in the process
+                of retrieving field values from the OCR output string - not particularly useful for this ID context.
+            id_string (str): A string containing some ID information.
+            barcode_data (dict, Optional): A dictionary object containing information extracted from a barcode.
+            fuzzy_min_ratio (float): The threshold ratio for a minimum, acceptable ratio of fuzziness when comparing
+                two strings.
+            max_multi_line (int): Specifies the maximum number of lines that is to be extracted from fields that are
+                noted as running onto multiple lines.
+
+        Returns:
+            (dict): A dictionary object containing the relevant, extracted ID information.
+        """
+        id_info = {}
+        regexp = re.compile(r'[0-9]{6,10}')
+        # Check if barcode data is available.
+        if barcode_data and barcode_data['identity_number']:
+            id_info['identity_number'] = barcode_data['identity_number']
+        for line_index, line in enumerate(id_string.split('\n')):
+            is_match = re.match(regexp, re.sub('[^\d]', '', line))
+            # Check if ID number was already extracted from barcode data.
+            if is_match:
+                if 'identity_number' not in id_info:
+                    # Populate id_info with the student/staff number.
+                    id_info['identity_number'] = re.sub('[^\d]', '', line)
+                # Retrieve some more information from the previous line.
+                if line_index - 1 >= 0:
+                    try:
+                        # Split the line on spaces.
+                        id_line = id_string.split('\n')[line_index - 1].split(' ')
+                        # Attempt to extrapolate sex.
+                        sex = 'M' if id_line[0] == 'Mr' else None
+                        sex = 'M' if id_line[0] == 'Ms' else sex
```

Comment:
Shouldn't it be
```python
sex = 'F' if id_line[0] =='Ms' else sex
```
?

Reply:
It should indeed. Nice catch. Apologies are in order once more.